app/controllers/vendor_sweets_controller.rb

```diff
@@ -0,0 +1,21 @@
+class VendorSweetsController < ApplicationController
+  
+  def create
+       @vendor_sweet = VendorSweet.new(vendor_sweet_params)
+    if @vendor_sweet.save
+     redirect_to vendor_path(@vendor_sweet.vendor)
+    else
+       @vendor = Vendor.find_by(id: vendor_sweet_params[:vendor_id])
+       
+       @sweets = Sweet.all
+       
+      render 'vendors/show'
```

Comment:
Remember, only define what you need! You're not using these instance variables. 
```suggestion
      render 'vendors/show'
```